src/iop/highlights.c

```diff
@@ -531,55 +531,110 @@ static void process_lch_bayer(dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *pie
   }
 }
 
-static void process_lch_xtrans(
-    const void *const ivoid,
-    void *const ovoid,
-    const int width,
-    const int height,
-    const float clip,
-    const dt_iop_roi_t *const roi_in,
-    const uint8_t (*const xtrans)[6])
+static void process_lch_xtrans(dt_iop_module_t *self, const void *const ivoid,
+                               void *const ovoid, const dt_iop_roi_t *const roi_in,
+                               const dt_iop_roi_t *const roi_out, const float clip)
 {
+  const uint8_t (*const xtrans)[6] = self->dev->image_storage.xtrans;
+
 #ifdef _OPENMP
 #pragma omp parallel for schedule(static) default(none)
```

Comment:
Yes, i think it just means it was able to parallelize it better, so more threads were actually running => in summary it took more time.